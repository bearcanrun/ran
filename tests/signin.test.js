/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../pages/signin';

const props = {
  url: {
    query: {
      postId: '123456',
      postTitle: 'RAN test'
    },
    pathname: '/'
  },
  pathname: '/',
  actions: {
    logout: () => ''
  }
};

jest.mock('../components/LinkList', () => 'LinkList');

describe('SignIn page snapshot', () => {
  it('should render correctly', done => {
    const component = renderer.create(
      <SignIn url={props.url} actions={props.actions} {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    setTimeout(() => done());
  });
});
