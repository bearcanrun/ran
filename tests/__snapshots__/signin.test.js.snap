// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn page snapshot should render correctly 1`] = `
<div
  className="sc-bdVaJa iNGvMq"
>
  <header
    className="sc-gzVnrw gqinNP"
  >
    <img
      alt="RAN!"
      className="sc-htoDjs eALRuc"
      src="https://user-images.githubusercontent.com/694940/29736531-6ab509e8-8a02-11e7-8e61-66e5ea4e29b8.png"
    />
    <LinkList
      authenticated={false}
      logout={[Function]}
      pathname="/"
    />
  </header>
  <div
    className="sc-dnqmqq cxGlgF"
  >
    <p>
      Example app of RAN! In the interest of fostering an open and welcoming environment, we as contributors, maintainers and writers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation. Please be kind and careful when you write something here. It is just showing how RAN! works on production. These posts are not related to RAN, Author of RAN, Contributors and Maintainers. Thank you for being kind, funny and awesome. For details, Please
       
      <a
        href="https://github.com/Sly777/ran/blob/master/CODE_OF_CONDUCT.md"
        rel="noopener noreferrer"
        target="_blank"
      >
        click here.
      </a>
    </p>
  </div>
  <div>
    <div
      className="sc-EHOje hnaVRQ"
    >
      <h1>
        Sign In
      </h1>
      <form>
        <div>
          <input
            name="email"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            type="email"
          />
          <div />
        </div>
        <div>
          <input
            name="password"
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
          />
          <div />
        </div>
        <br />
        <button
          className="sc-bZQynM cuvfAq"
          disabled={true}
          onClick={[Function]}
        >
          Sign In
        </button>
      </form>
    </div>
    <br />
    <div />
  </div>
</div>
`;
